### PR TITLE
Change target SDK version to 33

### PR DESCRIPTION
**Pull request by jpvasconcellos**
#80

Updating target sdk version

### CODE REVIEW

**Comment by jpvasconcellos** [comment]
Tomorrow morning I'll revert the changes in the auto-generated documentation and bump the other samples

File: zello-sdk-sample-contacts/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
```
Comment (zello-eugene) [inline zello-sdk-sample-contacts/gradle/wrapper/gradle-wrapper.properties:6]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
```

File: zello-sdk-sample-ptt/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
```
Comment (zello-eugene) [inline zello-sdk-sample-ptt/gradle/wrapper/gradle-wrapper.properties:5]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
```

File: zello-sdk-sample-signin/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
```
Comment (zello-eugene) [inline zello-sdk-sample-signin/gradle/wrapper/gradle-wrapper.properties:5]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
```

File: zello-sdk-sample/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
```
Comment (zello-eugene) [inline zello-sdk-sample/gradle/wrapper/gradle-wrapper.properties:5]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
```

File: zello-sdk/src/main/java/com/zello/sdk/headset/HeadsetMediaSessionImpl.kt [hunk 5]
```diff
@@ -39,8 +39,9 @@ class HeadsetMediaSessionImpl : HeadsetMediaSession {
 			val mediaButtonIntent = Intent(Intent.ACTION_MEDIA_BUTTON).run {
 				setClass(context.applicationContext, MediaButtonReceiver::class.java)
 			}
-			session.setMediaButtonReceiver(PendingIntent.getBroadcast(context.applicationContext, 0, mediaButtonIntent, 0))
-			session.isActive = true;
+			val intentFlag = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) PendingIntent.FLAG_IMMUTABLE else 0
```
Comment (zello-eugene) [inline zello-sdk/src/main/java/com/zello/sdk/headset/HeadsetMediaSessionImpl.kt:42]: It's Android M, not S, that introduced `FLAG_IMMUTABLE`. Ultimately, we should bump the min version to M anyway, which will remove the need to conditionally apply the flag.
```suggestion
			val intentFlag = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) PendingIntent.FLAG_IMMUTABLE else 0
```

File: zello-sdk-sample-contacts/build.gradle [hunk 6]
```diff
@@ -23,13 +23,13 @@ java {
...
 
 android {
-	compileSdkVersion 30
-	buildToolsVersion "31.0.0"
+	compileSdkVersion 33
+	buildToolsVersion '33.0.0'
 
 	defaultConfig {
 		applicationId "com.zello.sdk.sample.contacts"
 		minSdkVersion 16
```
Comment (zello-eugene) [inline zello-sdk-sample-contacts/build.gradle:31]:
```suggestion
		minSdkVersion 23
```

File: zello-sdk-sample-misc/build.gradle [hunk 7]
```diff
@@ -17,13 +17,13 @@ repositories {
 apply plugin: "com.android.application"
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 33
     buildToolsVersion "31.0.0"
 
     defaultConfig {
         applicationId "com.zello.sdk.sample.miscellaneous"
         minSdkVersion 16
```
Comment (zello-eugene) [inline zello-sdk-sample-misc/build.gradle:25]:
```suggestion
        minSdkVersion 23
```

File: zello-sdk-sample-ptt/build.gradle [hunk 8]
```diff
@@ -17,13 +17,13 @@ repositories {
 apply plugin: "com.android.application"
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 33
     buildToolsVersion "31.0.0"
 
     defaultConfig {
         applicationId "com.zello.sdk.sample.ptt"
         minSdkVersion 16
```
Comment (zello-eugene) [inline zello-sdk-sample-ptt/build.gradle:25]:
```suggestion
        minSdkVersion 23
```

File: zello-sdk-sample-signin/build.gradle [hunk 9]
```diff
@@ -17,13 +17,13 @@ repositories {
 apply plugin: "com.android.application"
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 33
     buildToolsVersion "31.0.0"
 
     defaultConfig {
         applicationId "com.zello.sdk.sample"
         minSdkVersion 16
```
Comment (zello-eugene) [inline zello-sdk-sample-signin/build.gradle:25]:
```suggestion
        minSdkVersion 23
```

File: zello-sdk-sample/build.gradle [hunk 10]
```diff
@@ -21,13 +21,13 @@ apply plugin: "com.android.application"
 apply plugin: "kotlin-android"
 
 android {
-	compileSdkVersion 30
+	compileSdkVersion 33
 	buildToolsVersion "31.0.0"
 
 	defaultConfig {
 		applicationId "com.zello.sdk.sample"
 		minSdkVersion 16
```
Comment (zello-eugene) [inline zello-sdk-sample/build.gradle:29]:
```suggestion
		minSdkVersion 23
```

File: zello-sdk/build.gradle [hunk 11]
```diff
@@ -141,13 +141,13 @@ tasks.whenTaskAdded { task ->
...
-	buildToolsVersion "31.0.0"
+	compileSdkVersion 33
+	buildToolsVersion '33.0.0'
 
 	ext.sdkVersion = "4.116.0"
 
 	defaultConfig {
-		targetSdkVersion 30
+		targetSdkVersion 33
 		minSdkVersion 16
```
Comment (zello-eugene) [inline zello-sdk/build.gradle:151]:
```suggestion
		minSdkVersion 23
```

**Review by zello-eugene — COMMENTED** [review]
Would you also update the min SDK version to 16 all around, as suggested in comments below, and use the same Gradle 5.2.1 in all projects. Less importantly, I personally prefer to use `"` universally instead of `'` as it ends up mixed up and looking untidy, but maybe that's just me LOL.

**Review by zello-eugene — APPROVED** [review]
👍

**Pull request merged**